### PR TITLE
Added Image Download Button

### PR DESCRIPTION
**Pull request by abdullahranginwala**
# Description

Once all the operations are processed, an image download window is popped again and again. Instead, a download button is added so the user can download the processed image without executing the operations again.
- Feature: #118 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

Also, include screenshots for verification and reviewing purpose.

https://user-images.githubusercontent.com/19731056/224710947-d4c94eb1-17b7-4ff8-beac-ef8ea573d18d.mov

# Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my code
- [ x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ghost** [comment]
i wanna work on this issue

**Comment by kaveeshadinamidu** [comment]
@abdullahranginwala When we are executing the pipeline it always shows the image save option. So by this issue what we are expecting is removing that image save option dialog and adding a separate image download option. So your approach is fine. But need some modifications to do. Add that image download button under the preview panel.

As well as don't try to fix several issues in one PR. I see that you have added undo and redo functions also. Try to fix one issue with one PR. 

Thank you!

File: imagelab_electron/index.html [hunk 1]
```diff
@@ -213,6 +213,26 @@
                       vertical-align: super;
                     "
                   ></span>
+                  <span
+                    onclick="undo()"
```
Comment (kaveeshadinamidu) [inline imagelab_electron/index.html]: Undo and redo functions need to be done in a separate PR.

File: imagelab_electron/workspace.js [hunk 2]
```diff
@@ -89,6 +90,16 @@ workspace.addChangeListener(function (event) {
   }
 });
 
+/**
+ * This method handles the undo function
+ */
+function undo() {
```
Comment (kaveeshadinamidu) [inline imagelab_electron/workspace.js]: These functions need to implemented in a separate PR.

**Review by kaveeshadinamidu — CHANGES_REQUESTED** [review]

**Comment by abdullahranginwala** [comment]
Thank you for your feedback! I have already made a PR for the undo and redo function which has been merged. There are some sync issues with my local repository. I'll fix that and implement the other feedback as well

**Comment by abdullahranginwala** [comment]
@kaveeshadinamidu I've implemented the feedback but I'm not sure how I can remove the 'undo/redo feature' commit. The PR for that particular issue has already been merged into the master branch but it got added here due to some merge conflicts. I've tried to clean up and fix the conflict. Would you recommend that I squash all the commits or is this good?

**Comment by 1Shubham7** [comment]
I want to  work on this issue.

File: imagelab_electron/index.html [hunk 3]
```diff
@@ -273,6 +273,18 @@
...
             </div>
           </div>
+          <div class="download-button">
+            <span
+              onclick="downloadImage()"
+              title="Download"
+              class="mdi mdi-download preview-pane"
+              style="
+                font-size: 30px;
+                color: #000000;
```
Comment (SahanDisa) [inline imagelab_electron/index.html:283]: Do you have any preferred reason to use this colour?

File: imagelab_electron/styles/style.css [hunk 4]
```diff
@@ -65,6 +65,21 @@ body {
   font-weight: bold;
 }
 
+.download-button {
```
Comment (SahanDisa) [inline imagelab_electron/styles/style.css:68]: Further let's improve the UX

**Review by SahanDisa — APPROVED** [review]

**Pull request merged**